### PR TITLE
feat: Add the wallet address in the account page

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR does the following:
- It adds a the address and a copy button for its, below the username in the accounts page.
- It abstracts the concept of timing stuff into a hook.

Desktop version:
copy-feature

Mobile version:
copy-mobile

### CODE REVIEW

File: webapp/src/modules/translation/locales/zh.json [hunk 1]
```diff
@@ -257,6 +257,9 @@
     "invalid_owner": "您不是{name}的持有者。",
     "submit": "确认"
   },
+  "account_page": {
+    "copied": "已复制 :heavy_check_mark:"
```
Comment (nachomazzara) [inline webapp/src/modules/translation/locales/zh.json:261]: missing emoji?

File: webapp/src/lib/timer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { renderHook, act } from '@testing-library/react-hooks'
```
Comment (nachomazzara) [inline webapp/src/lib/timer.spec.ts:1]: ❤️

File: webapp/src/modules/translation/locales/zh.json (hunk 1 above)
Comment (LautaroPetaccio) [reply webapp/src/modules/translation/locales/zh.json:261]: I've replicated the same line that's on the settings_page translations ([here](https://github.com/decentraland/marketplace/pull/351/files/5c2272e636bd59443e83e12dd64abc6ae0b7582e#diff-7ca97b120f16479fe42c01573b822edbb01297bfb74c09dd6a0c8fd46b91d2c9R267)).
But upon checking it, it seems to show the text `:heavy_check_mark:`. Should I change it in all the translations?

**Review by cazala — APPROVED** [review]
LGTM! There seems to be a conflict on the `package.json` tho, resolving that we're good to go

File: webapp/src/components/AccountPage/AccountPage.tsx [hunk 3]
```diff
@@ -1,19 +1,62 @@
...
 import { NFTBrowse } from '../NFTBrowse'
 import { Navigation } from '../Navigation'
 import { NavigationTab } from '../Navigation/Navigation.types'
 import { locations } from '../../modules/routing/locations'
-import { View } from '../../modules/ui/types'
 import { Props } from './AccountPage.types'
 import { Column } from '../Layout/Column'
 import './AccountPage.css'
 
+const copyAddressDesktop = (
```
Comment (nicosantangelo) [inline webapp/src/components/AccountPage/AccountPage.tsx:22]: In general, if a function is returning a JSX element, it should be a Component. This helps avoid a few problems like hooks not working and keeps the code a bit more consistent. Same goes for `copyAddressMobile` below!

Moreover, extracting this code is OK, but keeping with our own guidelines we tend to keep the JSX on the component that's using it **unless** it's needed in 2+ places (we could argue that some pieces of code are repeated but extracting them is overkill). IF we do extract it, we try to keep it in it's own folder+files.

Again, nothing wrong with the code per-se

File: webapp/src/components/AccountPage/AccountPage.tsx [hunk 4]
```diff
@@ -1,19 +1,62 @@
...
+  address: string,
+  hasCopiedAddress: boolean,
+  handleCopying: () => void
+) => (
+  <div className="profile-copy-text-mobile">
+    <CopyToClipboard text={address} onCopy={handleCopying}>
+      {hasCopiedAddress ? (
+        <span className="copied">{t('settings_page.copied')}</span>
+      ) : (
+        <span className="copy">{t('settings_page.copy_address')}</span>
```
Comment (nicosantangelo) [inline webapp/src/components/AccountPage/AccountPage.tsx:54]: I think the I18n key here should be account_page right? we could also move it to a common key like global or the like

**Review by nicosantangelo — CHANGES_REQUESTED** [review]
Awesome work 🔥

File: webapp/src/components/AccountPage/AccountPage.tsx [hunk 5]
```diff
@@ -27,6 +70,7 @@ const AccountPage = (props: Props) => {
   const isCurrentAccount =
     address === undefined || (wallet && wallet.address === address)
 
+  const [hasCopiedAddress, handleCopying] = useTimer(1200)
```
Comment (nicosantangelo) [inline webapp/src/components/AccountPage/AccountPage.tsx:73]: Sorry, one more thing, following the React hook naming conventions, the method should be called `setHasCopiedAddress` like it was called [here](https://github.com/decentraland/marketplace/pull/351/files#diff-27bae567c208e7553e5d83ce3d9376e8efb65574c4aef007dcc4e56a319c3bcfL31)

same goes for the line [here](https://github.com/decentraland/marketplace/pull/351/files#diff-27bae567c208e7553e5d83ce3d9376e8efb65574c4aef007dcc4e56a319c3bcfR32)

Comment (cazala) [reply webapp/src/components/AccountPage/AccountPage.tsx:73]: that would be right for a `useState` hook but this is a custom hook: https://github.com/decentraland/marketplace/pull/351/files#diff-ce54687ac19194251ade742703e87bf50231b4061c93dd39ffc1fa60adaf7b3cR18

the second deconstructed prop is the `startTimer`, not a setter.

File: webapp/src/components/AccountPage/AccountPage.tsx [hunk 6]
```diff
@@ -66,6 +110,13 @@ const AccountPage = (props: Props) => {
               <div className="profile-name">
                 <Profile address={address} textOnly inline={false} />
               </div>
+              <div className="profile-address">
+                <div className="profile-address-hash">{address}</div>
+                {!isMobile() &&
+                  copyAddressDesktop(address, hasCopiedAddress, handleCopying)}
+              </div>
+              {isMobile() &&
```
Comment (cazala) [inline webapp/src/components/AccountPage/AccountPage.tsx:118]: could we use a ternary here? we are calling `isMobile()` twice, also to keep the code more dry

Comment (LautaroPetaccio) [reply webapp/src/components/AccountPage/AccountPage.tsx:118]: The thing is that the the button is placed in a different place if the page is viewed in mobile or desktop, so if we used only one ternary, we would have to repeat part of the code.

If you think that it's fine to repeat that section, I can change it in a heart beat.

File: webapp/src/components/AccountPage/AccountPage.tsx (hunk 5 above)
Comment (LautaroPetaccio) [reply webapp/src/components/AccountPage/AccountPage.tsx:73]: That's correct, but we can think the `useTimer` as a normal hook, as it sets what it returns. It's true that it's not exporting the exact method that changes the value, but we can treat it as if it does.

Both names are correct though, whichever you think is right works for me.

File: webapp/src/components/AccountPage/AccountPage.tsx (hunk 6 above)
Comment (cazala) [reply webapp/src/components/AccountPage/AccountPage.tsx]: ahh you are right, mala mia, i didn't see the `<div/>` 😅

File: webapp/src/lib/timer.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    undefined
+  )
+
+  const startTimer = useCallback(() => {
+    if (timeoutId) {
+      clearTimeout(timeoutId)
+    }
+    setRunningStatus(true)
+    const newTimeoutId = setTimeout(() => setRunningStatus(false), delay)
+    setTimeoutId(newTimeoutId)
```
Comment (cazala) [inline webapp/src/lib/timer.ts:15]: This is super minor, but you don't really need a re-render of the component when you modify the `timeoutId` (unlike the `isRunning` which does require a re-render when modified). So you could use a mutable reference here, which is what's usually used for timeoutIds: https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

Comment (LautaroPetaccio) [reply webapp/src/lib/timer.ts:15]: Great catch! I'll check that out to see if I can implement it.

**Review by nicosantangelo — APPROVED** [review]

**Pull request merged**